ribbon-loadbalancer/src/main/java/com/netflix/loadbalancer/BaseLoadBalancer.java

```diff
@@ -209,18 +210,30 @@ public void initWithNiwsConfig(IClientConfig clientConfig) {
                 .getProperty(CommonClientConfigKey.NFLoadBalancerRuleClassName);
         String pingClassName = (String) clientConfig
                 .getProperty(CommonClientConfigKey.NFLoadBalancerPingClassName);
-
         IRule rule;
         IPing ping;
+        LoadBalancerStats stats;
         try {
             rule = (IRule) ClientFactory.instantiateInstanceWithClientConfig(
                     ruleClassName, clientConfig);
             ping = (IPing) ClientFactory.instantiateInstanceWithClientConfig(
                     pingClassName, clientConfig);
+            stats = createLoadBalancerStatsFromConfig(clientConfig);
         } catch (Exception e) {
             throw new RuntimeException("Error initializing load balancer", e);
         }
-        initWithConfig(clientConfig, rule, ping);
+        initWithConfig(clientConfig, rule, ping, stats);
+    }
+
+    private LoadBalancerStats createLoadBalancerStatsFromConfig(IClientConfig clientConfig) {
+        String loadBalancerStatsClassName = clientConfig
+                .get(CommonClientConfigKey.NFLoadBalancerStatsClassName, LoadBalancerStats.class.getName());
+        try {
+            return (LoadBalancerStats) ClientFactory.instantiateInstanceWithClientConfig(
+                    loadBalancerStatsClassName, clientConfig);
+        } catch (Exception e) {
+            throw new RuntimeException("Error initializing LoadBalancerStats", e);

```

Comment:
instead of throwing a RuntimeException, it may be more compatible if we fallback to emitting a WARN log and create the stats class with the existing `new LoadBalancerStats(clientName)`.